### DISCUSSION TITLE
Idea of Mutable JobState

### DISCUSSION THREAD

**Discussion by drusellers** [quartznet/quartznet#2612] [Q&A]
Is there a Quartz native way to update Job State ?

The idea is that as a job progresses, I want to update some state. That way if the process restarts, it can pick back up where it left off.

**Comment by drusellers** [comment]
So, I was reading https://github.com/quartznet/quartznet/discussions/1823 

What I want is to do something like, this at the top of the job

```csharp
var currentPage = 1;
if (context.MergedJobDataMap.TryGetIntValue("CurrentPage", out var p))
{
    currentPage = p;
}
```

then at the bottom of the loop do

```csharp
context.Put("CurrentPage", currentPage);
// or
context.MergedJobDataMap.Put("CurrentPage", currentPage);
```

So, that if the process exits, and the job restarts it restarts from the correct spot.

**Reply by lahma** [reply]
For reading, `context.MergedJobDataMap` is ideal as you will get both trigger and job data from there, but for writing it should not be used. `JobDataMap goodForWrite = context.JobDetail.JobDataMap;` , this will have full interface for interacting.
